subsys/bluetooth/host/keys_br.c

```diff
@@ -52,31 +59,64 @@ struct bt_keys_link_key *bt_keys_get_link_key(const bt_addr_t *addr)
 	}
 
 	key = bt_keys_find_link_key(BT_ADDR_ANY);
+#if IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+	if (!key) {
+		int i;
+
+		for (i = 1; i < ARRAY_SIZE(key_pool); i++) {
+			struct bt_keys_link_key *current = &key_pool[i];
+
+			if (current->aging_counter < key->aging_counter) {
+				key = current;
+			}
+
+			bt_keys_link_key_clear_addr(&key->addr);
```

Comment:
This cannot be right. The clear function is in the for-loop. 

I suggested earlier and I'll suggest it again. Maybe just remove supporting the BT_KEYS_OVERWRITE_OLDEST option for now. If all you need is saving the key. It can be added later.